vispy/visuals/mesh.py

```diff
@@ -545,6 +527,19 @@ def _update_data(self):
                 colors = self._color.rgba
         self.shared_program.vert['position'] = self._vertices
 
+        # Update buffer of texture coordinates.
+        if self.shading == 'smooth' and not md.has_face_indexed_data():
+            texcoords = md.get_texcoords()
+        else:
+            texcoords = md.get_texcoords(indexed='faces')
+        if texcoords is not None:
+            self._texcoords.set_data(texcoords, convert=True)
+            pass_texcoord = Function("void pass_tc() { $output = $input; }")
+            pass_texcoord['input'] = self._texcoords
+            pass_texcoord['output'] = self.texcoord_varying
+            vert_pre = self._get_hook('vert', 'pre')
+            vert_pre.add(pass_texcoord())
```

Comment:
`texcoords` are passed throught the vertex shader only if texcoords are defined in the MeshVisual